docs: Desktop access: Move local Windows users content changes to a new version of the gett… (#31678)
gravitational/teleport: docs/pages/desktop-access/getting-started.mdx

---
title: Configure access for local Windows users

> description: Configure passwordless access for local Windows users on Windows computers that aren't joined to an Active Directory domain.

Reviewer: ```suggestion
description: Configure passwordless access for local Windows users.
```

Technically it does work on domain-joined hosts too.

videoBanner: 9DyKQbg4ORc
---

This guide demonstrates how to configure Teleport to provide secure, passwordless access

> to Microsoft Windows desktops that aren't joined to an Active Directory domain.

Reviewer: Is the description rendered on the page already? If it is, we probably don't need to say this twice.

Author: The description frontmatter is used by web crawlers, not rendered as text on the page (that's probably configurable but not typical).

The following diagram provides a simplified view of how you configure Teleport
 for local Windows users:

![Configure access for local Windows users](../../img/desktop-access/windows-local-users.png)

## Prerequisites

To complete the steps in this guide, verify your environment meets the following requirements:

- Access to a running Teleport cluster.
- A Linux server to run the Teleport Windows Desktop Service.
  You can reuse an existing server running any other Teleport instance. 
  However, you must generate a new invitation token to add the Windows Desktop Service
  to an existing Teleport instance.
- A physical or virtual machine running Microsoft Windows with Remote Desktop enabled 

>   and the RDP port—typically, port 3389—open for inbound connections from the Linux server.

Reviewer: ```suggestion
  and the RDP port (typically 3389) open for inbound connections from the Linux server.
```

- (!docs/pages/includes/tctl.mdx!)

## Step 1/4. Prepare Windows

Before you can enroll a Microsoft Windows desktop as a resource to be protected by Teleport,
you need to prepare the Microsoft Windows environment with a certificate from the Teleport 
certificate authority (CA) and modify the default authentication service for remote desktop
access.

To prepare a Windows computer:

1. Open a Command Prompt (`cmd.exe`) window on the Windows computer you want to enroll.

> 1. Export the Teleport user certificate authority by running the following command:

Reviewer: ```suggestion
1. Export the Teleport user certificate authority by running the following command, replacing teleport-proxy.example.com with the address of your proxy server:
```

(or use a Var component)

   ```code
   $ curl -o teleport.cer https://teleport-proxy.example.com/webapi/auth/export?type=windows
   ```

1. Download the Teleport Windows Auth [setup program](https://cdn.teleport.dev/teleport-windows-auth-setup-v(=teleport.version=)-amd64.exe).

1. Doubleclick the executable you downloaded to run the Teleport Windows Auth Setup program 
interactively and select the Teleport certificate that you exported when prompted.

   The setup program:

>    - Installs the selected Teleport certificate.

Reviewer: ```suggestion
   - Configures Windows to trust the Teleport CA
```

>    - Loads the required dynamic link library (DLL) for Teleport to use.

Reviewer: ```suggestion
   - Installs the required dynamic link library (DLL) for Teleport to use.
```

   - Disables Network Level Authentication (NLA) for remote desktop services. 

1. Restart the computer.

> After you download the setup program so it's available on the Windows computer, 

Reviewer: I don't think we need to say 'after you downloaded the program' - we already told the user to download it and double click it.

We probably just want to say something like "If you prefer an automated install, you can invoke the installer from an administrative command prompt"

1. Sign in to the Teleport cluster from your administrative workstation to authenticate your identity.

1. Generate a short-lived invitation token by running the following command:

   ```code
   $ tctl tokens add --type=windowsdesktop
   ```

   The command displays information similar to the following:

   ```text
   The invite token: (=presets.tokens.first=)
   This token will expire in 60 minutes.
   ```

1. Copy the token from your administrative workstation and paste it into a file on the 
Linux server where you want to run the Windows Desktop Service.

   For example, create a file named `/tmp/token` on the Linux server and paste the token
   into the file.

1. Check whether Teleport is installed on the Linux server where you want to run the 
Windows Desktop Service by running the following command:

   ```code
   $ teleport version
   ```

   If Teleport isn't installed on the Linux server, follow the appropriate [Installation
instructions](../installation.mdx#installation-instructions) for your environment.

1. Configure the `/etc/teleport.yaml` file on the Linux server with settings similar to the 
following for the Windows Desktop Service: 

   ```yml

>    version: v3

Reviewer: YAML is pretty sensitive to whitespace. I see you indented this a bit make it "under" this step in the ordered list, but I just want to make sure that when this renders there won't be 3 leading spaces.

Author: It seems like the copy button strips out the extra space.

   teleport:
     nodename: <Var name="windows.teleport.example.com" />

>      proxy_server: <Var name="teleport.example.com:443" />

Reviewer: Whatever we do, let's make the default value for proxy_server here match what was used in the curl command above.

     auth_token: <Var name="/tmp/token" />
   windows_desktop_service:
     enabled: yes
     non_ad_hosts:
     - 192.0.2.156

>    auth_service:

Reviewer: ```suggestion

   auth_service:
```

I find it helpful to separate the block that is just disabling other services.

     enabled: no
   proxy_service:
     enabled: no
   ssh_service:
     enabled: no
   ```

   In this file:

   - Set the `proxy_server` to the address of your Teleport cluster.

>    - List the Windows IP addresses or DNS names under `non_ad_hosts`.

Reviewer: ```suggestion
   - List the Windows IP addresses or hostnames under `non_ad_hosts`.
```

DNS is the mechanism by which a hostname is resolved to an IP address, so I would prefer using hostname over "DNS name"

   Teleport can't automatically discover Windows desktops without Active Directory. 
   Therefore, you must specify the Windows IP addresses or DNS names that you want to 
   enroll using the Teleport configuration file or use the Teleport API  
   to build your own integration. You can find an example of API integration on 
   [GitHub](https://github.com/gravitational/teleport/tree/master/examples/desktop-registration).

1. (Optional) Add labels for the Windows Desktop Service to the configuration file.

>    You can attach labels to Windows computers by matching to their hostnames.

Reviewer: We have a whole separate section on labeling in the RBAC page, does it make sense to link that here?

   For example, the following adds the `cloud: ec2` label to computers that have 

>    EC2 private IP DNS names matching `.us-east-2.compute.internal`:

Reviewer: ```suggestion
   hostnames ending with `.us-east-2.compute.internal`:
```

"EC2 private IP DNS names" is a mouthful.

   teleport:
     nodename: windows.teleport.example.com
     proxy_server: teleport-proxy.example.com:443
   windows_desktop_service:
     enabled: yes
     non_ad_hosts:
   -    - 192.0.2.156
   +    - ip-192-0-2-156.us-east-2.compute.internal
   + host_labels:
   +    - match: '.*\.us-east-2.compute.internal'
   +      labels:
   +        cloud: ec2
   ```

1. (!docs/pages/includes/start-teleport.mdx service="the Teleport Windows Desktop Service"!)

## Step 3/4. Configure remote Windows desktop access

To access a remote desktop, a Teleport user must have a role with the
appropriate permissions for that desktop.

To configure a role for desktop access:

1. Sign in to the Teleport cluster from your administrative workstation to authenticate your identity.

1. Create a `windows-desktop-admins.yaml` file to define a new role:

   ```yaml
   kind: role
   version: v6
   metadata:
     name: windows-desktop-admins
   spec:
     allow:
       windows_desktop_labels:
         "*": "*"
       windows_desktop_logins: ["Administrator", "alice"]
   ```

   In this file:

   - Set `windows_desktop_labels` to specify any labels you want to use to allow or deny access to specific hosts.

>    - Set `windows_desktop_logins` to specify the Windows logins that Teleport users assigned to the role

Reviewer: Maybe we should also expand on automatic user creation (the logins you specify in `windows_desktop_logins` don't have to exist ahead of time on the Windows machine, Teleport can create them on demand).

This is only true for non-AD hosts and only if RBAC allows it, and it's not well documented.
